### ISSUE TITLE
Write operations don't complete if window is reloaded just after

### ISSUE THREAD

**Issue by dphochman** [cyrilletuzi/angular-async-local-storage#162]
When we reload after clearing or writing to local storage, the item is not written to local storage:

```typescript
public async clear_save_reload(key: string, value: any): Promise<void> {
   await this.localStorage.clear().toPromise();
   await this.localStorage.set(key, value).toPromise();
   window.reload();
}
```

According to https://stackoverflow.com/questions/49766364/indexeddb-flush-to-disk-on-chrome, Chrome may have an issue where you must wait for the IndexedDB operation to 'complete'.

The source code for [indexeddb-database.ts](https://github.com/cyrilletuzi/angular-async-local-storage/blob/master/projects/ngx-pwa/local-storage/src/lib/databases/indexeddb-database.ts) seems to show that `set()` waits for the 'success' event, instead of 'complete', and 'clear()' waits for the first event which may just be 'success' and not 'complete'.

This issue may or may not be related to [Issue #47](https://github.com/cyrilletuzi/angular-async-local-storage/issues/47). 

We are trying to work around this by introducing a delay-promise after each operation, but that doesn't seem optimal.

This package has been great, and I hope this report helps make it even better.
Thank you!

**Comment by cyrilletuzi** [comment]
Hi @dphochman, thanks for your feedback.

I'll do a test with the `complete` event when I have some time, but [MDN is saying](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API/Using_IndexedDB) that even the `complete` event may be too early:

> Note: As of Firefox 40, IndexedDB transactions have relaxed durability guarantees to increase performance (see bug 1112702.) Previously in a readwrite transaction IDBTransaction.oncomplete was fired only when all data was guaranteed to have been flushed to disk. In Firefox 40+ the complete event is fired after the OS has been told to write the data but potentially before that data has actually been flushed to disk. The complete event may thus be delivered quicker than before, however, there exists a small chance that the entire transaction will be lost if the OS crashes or there is a loss of system power before the data is flushed to disk. Since such catastrophic events are rare most consumers should not need to concern themselves further. If you must ensure durability for some reason (e.g. you're storing critical data that cannot be recomputed later) you can force a transaction to flush to disk before delivering the complete event by creating a transaction using the experimental (non-standard) readwriteflush mode (see IDBDatabase.transaction.

**Comment by cyrilletuzi** [comment]
Hello @dphochman, I tried to reproduce your issue with the same code, but it works for me.

So I need further details:
- what is your version of Chrome?
- what is your Angular version?
- what is your version of this lib?
- do you have a minimal reproduction of your issue?

**Comment by cyrilletuzi** [comment]
Hello @dphochman,

Even if I can't reproduce the issue for now, I suppose managing `complete` transactions event instead of `success` requests event is better.

So I did #169, that you can test with: `npm install @ngx-pwa/local-storage@beta`

Can you tell me if it fixes your issue?

Nevertheless, I still need the answers to my previous questions and a reproduction, as I need to add a test to check if it's really OK and that it will stay so in the future.

**Comment by cyrilletuzi** [comment]
Hello @dphochman,

The issue has been closed automatically as the PR has been merged and release in v8.2, but it would still be better if I can reproduce the issue to add a test to avoid regressions.

**Comment by sebastian-zarzycki-es** [comment]
I might or might not bump into something related. In my Angular app, I'm storing the logged in user's token using @ngx-pwa/local-storage. During the development, my application often reloads, because it has hot reloading configured, so whenever I'm changing something in the code, it reloads in the browser. 

I've noticed that every once a while, after such reload, I'm no longer logged in as a user I was just a moment ago. I have no reason to assume it is an error in my own code or on the server code. It looks to me, like the login token just wasn't saved and after a reload, it uses the old token to validate with server again. Which would be very weird, because I'm clearing the token in storage upon logout...

Anyway, I'm going to test it more, to see if I can pinpoint the issue. But local storage / indexeddb would be my first culprit in this case. I'm using 7.4.2 and Firefox 70, by the way.
